Slay the Spire fails to start

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you checked the wiki for solutions?

- [X] I have searched the wiki

### Is this a Steam Snap-specific issue?

- [X] This is a Steam Snap-specific issue

### Current Behavior

Slay the Spire fails to start from Steam snap.

### Expected Behavior

Slay the Spire starts successfully.

### Steps To Reproduce

1. Install Slay the Spire
2. Press `Play`

### Environment

```yaml
os_release:
    name:               "Ubuntu"
    version:            "22.04.3 LTS (Jammy Jellyfish)"
snap_info:
    steam_revision:     154
    snapd_revision:     20092
lspci:
    00:02.0:            Intel Corporation TigerLake-LP GT2 [Iris Xe Graphics] (rev 01)
glxinfo:
    gpu:                Mesa Intel(R) Xe Graphics (TGL GT2)
    gpu_version:        4.6 (Core Profile) Mesa 23.1.7 - kisak-mesa PPA
lscpu:
    model_name:         11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu
```


### gaming-graphics-core22 version

kisak-fresh (default)

### Anything else?


Console logs when trying to launch StS:

```
19:51:53.367 INFO core.DisplayConfig> Reading info.displayconfig
19:51:53.370 INFO core.DisplayConfig> DisplayConfig successfully read.
19:51:53.370 INFO desktop.DesktopLauncher> [WINDOWED_MODE]
19:51:53.370 INFO desktop.DesktopLauncher> Settings successfully loaded
19:51:53.370 INFO desktop.DesktopLauncher> Launching application...
Exception in thread "LWJGL Application" java.lang.ExceptionInInitializerError
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setVSync(LwjglGraphics.java:558)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:124)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
	at org.lwjgl.opengl.LinuxDisplay.getAvailableDisplayModes(LinuxDisplay.java:954)
	at org.lwjgl.opengl.LinuxDisplay.init(LinuxDisplay.java:738)
	at org.lwjgl.opengl.Display.<clinit>(Display.java:138)
	... 2 more
AL lib: (EE) alc_cleanup: 1 device not closed
Uploaded AppInterfaceStats to Steam
```

Looks like `getAvailableDisplayModes` is returning an empty array, which is causing an index out of bounds error. This same exception happens on all versions of `gaming-graphics-core22`. I tried `kisak-turtle`, `kisak-fresh` and `oibaf-latest`.

Game works fine outside of snap.

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -378,6 +378,7 @@ parts:
       - psmisc
       - libfuse2:amd64
       - libfuse2:i386
+      - x11-xserver-utils
     override-build: |
       craftctl default
       # Set the snap version from the .deb package version

